cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerClusterHealthService.java

```diff
@@ -165,6 +208,26 @@ private static Optional<DetailedServicesHealthCheck> getDetailedServicesHealthCh
         return Optional.empty();
     }
 
+    private Set<String> collectDisconnectedNodeManagers(List<ApiRole> allNMRoles) {
+        return allNMRoles.stream()
+                .filter(this::isNodeManagerDisconnected)
+                .filter(nm -> nonNull(nm.getHostRef()))
+                .map(nm -> nm.getHostRef().getHostname())
+                .map(StringUtils::lowerCase)
+                .collect(Collectors.toUnmodifiableSet());
+    }
+
+    private boolean isNodeManagerDisconnected(ApiRole role) {
+        Predicate<ApiHealthCheck> disconnectedNMCheck = hc -> {
+            ApiHealthSummary summary = hc.getSummary();
+            String healthCheckName = hc.getName();
+            // The connectivty health check is disabled if the nodes are deocmmissioned
+            return NODE_MANAGER_CONNECTIVITY.equals(healthCheckName)
+                    && !DISABLED.equals(summary) && (BAD.equals(summary) || CONCERNING.equals(summary));
```

Comment:
Removed !DISABLED.equals(summary)

Reply:
An update on this. This is a crucial check to make sure fast upscale works. For hosts with the NodeManager role in STOPPED state, this heath check param is always DISABLED. Otherwise, it reports GOOD/BAD etc (depending on connectivity between NM and ResourceManager). A follow up to fix this has to be introduced

Reply:
Let `e1 = !DISABLED.equals(summary) && (BAD.equals(summary) || CONCERNING.equals(summary))` and `e2 = BAD.equals(summary) || CONCERNING.equals(summary)`. Then we can observe the following results depending on the value of `summary`:

* `summary == DISABLED`: `e1 == e2 == false`.
* `summary == BAD`: `e1 == e2 == true`.
* `summary == CONCERNING`: `e1 == e2 == true`.
* `summary` is some other value (so not `DISABLED`, `BAD` or `CONCERNING`): `e1 == e2 == false`.

Sorry but I do not see how the value of `summary == DISABLED` could influence the outcome here.